common/coin_support/eth.c

```diff
@@ -499,7 +507,7 @@ void eth_init_msg_data(MessageData *msg_data) {
 }
 
 int eth_byte_array_to_unsigned_txn(const uint8_t *eth_unsigned_txn_byte_array,
-                                   size_t byte_array_len, eth_unsigned_txn *unsigned_txn_ptr, const txn_metadata *metadata_ptr)
+                                   size_t byte_array_len, eth_unsigned_txn *unsigned_txn_ptr, txn_metadata *metadata_ptr)
```

Comment:
Why have we made a change in the function prototype when there's no change in the function?

Reply:
The function responsible for decoding payload now updates the decimal value on txn_metadata which is being called by this function.